### PR TITLE
Add GH Actions based CI

### PR DESCRIPTION
**Pull request by jaimergp**
Same approach as https://github.com/openmm/openmm-torch/pull/25

Actual checks show up in my fork until this is merged: https://github.com/jaimergp/openmm-plumed/pull/1

### CODE REVIEW

**Comment by jaimergp** [comment]
OpenCL tests now running here!

**Comment by jaimergp** [comment]
@peastman This is ready for review. It follows the same approach as https://github.com/openmm/openmm-torch/pull/25. We are lacking Python unit tests here, but I added an issue for that in #46 so we can address it in the future. 

As usual, the CI logs are not running here (yet), so you'll need to check the [mirrored PR in my fork](https://github.com/jaimergp/openmm-plumed/pull/1).

**Comment by peastman** [comment]
Looks good!

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    name: ${{ matrix.name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Linux CPU CUDA 10.0 Python 3.6
+            python-version: "3.6"
+            os: ubuntu-latest
+            gcc-version: "9"
+            cuda-version: "10.0"
```
Comment (raimis) [inline .github/workflows/CI.yml:29]: Is any reason why not to test with the latest CUDA?

Comment (jaimergp) [reply .github/workflows/CI.yml:29]: Not really. Installation size, if anything. We can change that or add multiple matrix entries for different CUDA versions.

Comment (raimis) [reply .github/workflows/CI.yml:29]: I think, only the latest CUDA version is enough. Also `conda install -c conda-forge openmm-plumed` installs CUDA 11.

Comment (jaimergp) [reply .github/workflows/CI.yml:29]: The CUDA versions depends on the conda version and the installed Nvidia drivers. Most of the time it'll be CUDA 11 but it's not guaranteed. Anyway, yes, I'll move this to 11.2.

**Comment by jaimergp** [comment]
See now @raimis -- we can only test against 11.0 because that's the latest available on CF. [Slowly getting there for 11.1 and 11.2, though](https://github.com/conda-forge/cudatoolkit-feedstock/issues/47). Notice that the Python tests are now running too!

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+jobs:
+  unix:
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Linux CPU CUDA 10.0 Python 3.6
```
Comment (raimis) [inline .github/workflows/CI.yml:25]: Sorry for being to pedantic, this still says "CUDA 10"

Comment (jaimergp) [reply .github/workflows/CI.yml:25]: Oops, good catch :D

**Comment by jaimergp** [comment]
I think we are ready to merge here @raimis @peastman !

**Review by raimis — APPROVED** [review]

**Comment by raimis** [comment]
@jaimergp thanks for your effort!

**Comment by raimis** [comment]
The tests have passed on `master`!

**Comment by jaimergp** [comment]
🎯 💯 Perfect score! Thanks for the merge!

**Pull request merged**